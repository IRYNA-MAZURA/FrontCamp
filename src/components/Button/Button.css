button {
  background-color: #555555;
  border: none;
  color: #FFFFFF;
  padding: 5px 20px;
  font-size: 18px;
  font-weight: 100;
  text-transform: uppercase;
  cursor: pointer;
  font-family: 'Montserrat', sans-serif;
}

button:active,
button:focus 
{
  outline: none;
}

.main {
  border-radius: 5px;
  font-size: 24px;
  padding: 10px 40px;
}

.left {
  border-radius: 3px 0 0 3px;
}

.right {
  border-radius: 0 3px 3px 0;
}

.active {
  background-color: #F65261;
}
