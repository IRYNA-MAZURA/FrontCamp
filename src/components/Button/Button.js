import React from 'react';

import './Button.css';

export const Button = ({type, isActive, returnType, text, handleClick = () => {}}) => {
    const isA = isActive ? ' active' : '';
    return (
        <button className={type + isA} onClick={() => handleClick(returnType)}>{text}</button>
    )
};
