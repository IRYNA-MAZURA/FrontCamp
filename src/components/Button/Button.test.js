import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { Button } from './Button';

const props = {
    type: 'main', 
    isActive: true,  
    text: 'text', 
    handleClick: jest.fn()
}

describe('Button component', () => {
    it('snapshot', () => { 
        const wrapper = renderer.create(<Button {...props} />).toJSON();       
        expect(wrapper).toMatchSnapshot();
    });

    it('action testing', () => {
        const wrapper = shallow(<Button {...props} />);
        wrapper.find('button').first().simulate('click');
        expect(props.handleClick).toHaveBeenCalled();
    });
});