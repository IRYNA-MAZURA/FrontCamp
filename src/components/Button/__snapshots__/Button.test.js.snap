// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component snapshot 1`] = `
<button
  className="main active"
  onClick={[Function]}
>
  text
</button>
`;
